graphql/src/store/resolver.rs

```diff
@@ -182,6 +183,30 @@ where
                 _ => true,
             }).unwrap_or(true)
     }
+
+    /// Compute special fields that are not stored such as as `entityCount`.
+    fn add_computed_fields(
```

Comment:
Overall, this is great, but I would do this part differently. Adding computed fields like this will mean that we'll always count entities for all subgraphs, even if `entityCount` is not part of the query.

What I would do instead is resolve fields like this lazily. Here's what I'd recommend:

1. Change the signature of `Resolver::resolve_scalar_value` so that it takes the object type and the field name:
   ```rust
   fn resolve_scalar_value(
       &self,
       object_type: &s::ObjectType,
       field_name: &q::Name,
       scalar_type: &s::ScalarType,
       value: Option<&q::Value>,
   ) -> q::Value {
   ```
2. Implement `resolve_scalar_value` in `StoreResolver`. Something like this should do:
   ```rust
    fn resolve_scalar_value(
        &self,	
        object_type: &s::ObjectType,
        field_name: &q::Name,
        scalar_type: &s::ScalarType,
        value: Option<&q::Value>,
    ) -> q::Value {
        match field_name.as_str() {
          "entityCount" => {
            ... parse subgraph ID, if possible, fall back to value ...
            ... obtain entity count ...
          },
          _ => value
        }.and_then(|value| value.coerce(scalar_type)).unwrap_or(q::Value::Null)
    }
   ```

Reply:
But I'm not completely convinced of my approach because it makes adding support for GraphQL interfaces harder (which don't have `ObjectType`). Perhaps let's go with your approach first.